ts/src/test/test.ts

```diff
@@ -397,12 +396,8 @@ export default class testMainClass extends baseMainTestClass {
                 }
             }
             // we don't throw exception for public-tests, see comments under 'testSafe' method
-            let failedMsg = '';
-            const errorsLength = errors.length;
-            if (errorsLength > 0) {
-                failedMsg = ' | Failed methods : ' + errors.join (', ');
-            }
-            dump (this.addPadding ('[INFO:PUBLIC_TESTS_END] ' + market['type'] + failedMsg, 25), exchange.id);
+            const failedMsg = errors ? errors.join (', ') : undefined;
+            dump (this.addPadding ('[INFO:PUBLIC_TESTS_END] ' + market['type'] + (errors ? (' | Failed methods : ' + failedMsg) : ''), 25), exchange.id);
```

Comment:
@carlosmiei I was actually trying to remove extra noise and make it inline, however, it seems to have ended up with more increased spaciousy code than before this PR :))
https://github.com/ccxt/ccxt/blob/master/ts/src/test/test.ts#L406-L413

Reply:
@ttodua yes, it broke the build. And besides that this is much harder to understand and easier to mess up so rule of thumb let's avoid convoluted statements